cpu/cortexm_common/cortexm_init.c

```diff
@@ -72,3 +72,40 @@ void cortexm_init(void)
     cortexm_init_isr_priorities();
     cortexm_init_misc();
 }
+
+static const uint32_t BFARVALID_MASK = (0x80 << SCB_CFSR_BUSFAULTSR_Pos);
+
+bool cpu_check_address(volatile const char *address)
+{
+#if defined(CPU_ARCH_CORTEX_M3) || defined(CPU_ARCH_CORTEX_M4) || \
+    defined(CPU_ARCH_CORTEX_M4F) || defined(CPU_ARCH_CORTEX_M7)
+    bool is_valid = true;
+
+    /* Clear BFARVALID flag */
+    SCB->CFSR |= BFARVALID_MASK;
+
+    /* Ignore BusFault by enabling BFHFNMIGN */
+    SCB->CCR |= SCB_CCR_BFHFNMIGN_Msk;
+    __asm volatile ("cpsid f;");
```

Comment:
AFAICT this does not inhibit interrupts, meaning the bus fault could possibly be caused by something else.

Reply:
"cpsid f" disables interrupts and all fault handlers, according to 3.10.2 of the Cortex-M3 Devices Generic User Guide.